pandas/core/groupby/groupby.py

```diff
@@ -1895,6 +1895,12 @@ def post_processor(vals: np.ndarray, inference: Optional[Type]) -> np.ndarray:
 
             return vals
 
+        if is_scalar(q):
+            q = np.array([q])
+
+        if not all(np.clip(q, 0, 1) == q):
```

Comment:
this validation shouldn't be here; we already have validations for non-groupby quantiles, I imagine we have tests for this already, so this might be duplicative, you'll have to trace.

Reply:
Please read the thread. Since it segfaults and jbrockmendel reproduced, obviously this isn't tested or validated in this case. Series/DataFrame.quantile calls NDFrame._check_percentils, but Groupby.quantile doesn't and it doesn't inherit from NDFrame either so the method is available.

Could:
- move the validation inside cython quantile, and delete NDFrame._check_percentils
- Or, ove NDFrame._check_percentile out to some util pyx and add a validation in Groupby.quantile. 

Either is fine by me.